src/libraries/Native/Unix/CMakeLists.txt

```diff
@@ -125,9 +135,8 @@ endif ()
 
 if (CMAKE_SYSTEM_NAME STREQUAL Linux)
    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_GNU_SOURCE")
-   set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} -Wa,--noexecstack")
-   set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--build-id=sha1")
-   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--build-id=sha1")
+   add_compile_options($<$<COMPILE_LANGUAGE:ASM>:-Wa,--noexecstack>)
+   add_link_options(-Wl,--build-id=sha1 -Wl,-z,relro,-z,now)
```

Comment:
relro and immediate-bind are ELF features, so might want to use them on all ELF targets instead?

Reply:
Also, for Darwin systems, you can use `-Wl,-bind_at_load`, which is roughly similar to immediate binding (`-z,now`).

Reply:
Another thing we might consider is, since we're using these two options and building stuff using position-independent code, to build everything with `-fno-plt`.  This should lead to slightly more efficient code when calling functions in shared libraries (PLT stubs aren't generated, and the GOT offets are used directly at each call site; this is exposed to optimizers, too, which might also result in better codegen for repeated calls to the same functions).

Reply:
Let's leave this possibility for another PR.